includes/class-amp-theme-support.php

```diff
@@ -397,12 +398,54 @@ public static function finish_init() {
 				self::redirect_non_amp_url( current_user_can( 'manage_options' ) ? 302 : 301 );
 			}
 
+			if ( MobileRedirectManager::is_enabled() && is_amp_available() ) {
+				$mobile_redirection_disabled = isset( $_GET[ MobileRedirectManager::NO_AMP_QUERY_VAR ] ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+				// If `noamp` query var is present, set a cookie to persist disabling of mobile redirection (if one has not already been set).
+				if ( $mobile_redirection_disabled && ! MobileRedirectManager::redirection_disabled() ) {
+					setcookie(
+						MobileRedirectManager::DISABLED_COOKIE_NAME,
+						'1',
+						[
+							'expires'  => 0, // Cookie will only last for the current browser session.
+							'path'     => '/',
```

Comment:
To account for multisite and subdirectory installs, this needs to use the home URL:

```suggestion
							'path'     => wp_parse_url( home_url( '/' ), PHP_URL_PATH ),
```